packages/pigeon/README.md

```diff
@@ -1,13 +1,354 @@
 # Pigeon
 
+// TODO: how to frame this?
+// To get started, see the Quickstart Guide in the [Example README](./example/README.md).
+
 Pigeon is a code generator tool to make communication between Flutter and the
 host platform type-safe, easier, and faster.
 
-Pigeon removes the necessity to manage strings across multiple platforms and languages.
-It also improves efficiency over common method channel patterns. Most importantly though,
-it removes the need to write custom platform channel code, since pigeon generates it for you.
+Pigeon works by reading special files which by convention are placed in a separate
+folder, `/pigeons`, alongside your `/lib` folder. You can put all of your definitions
+in a single file (e.g., `/pigeons/messages.dart`), or you can divide your methods
+and classes into separate files as you wish. After reading these files, Pigeon
```

Comment:
Using separate files has pitfalls and complexity; we shouldn't suggest it in the initial overview.

Reply:
Good to know, thanks.